### PR TITLE
Improved logging.

### LINKED ISSUE #195: Clean up app/setup logging.
**Describe the bug**
These logs should be under `DEBUG`:
```
[2020-03-05T00:51:04+08:00] INFO [rpc_server_02f9a04bf7d30c09c8b943735a1b3a1dc9f6ab9667bad1a2162a3df51cfce252be]: Inside RPC Accept on server side
[2020-03-05T00:51:04+08:00] INFO [rpc_server_02f9a04bf7d30c09c8b943735a1b3a1dc9f6ab9667bad1a2162a3df51cfce252be]: Reserving next ID on RPC Accept server side
[2020-03-05T00:51:04+08:00] INFO [rpc_server_02f9a04bf7d30c09c8b943735a1b3a1dc9f6ab9667bad1a2162a3df51cfce252be]: Accepting conn on RPC Accept server side
```

### LINKED ISSUE #196: Clean up logs on shutdown.
**Describe the bug**
```
[2020-03-05T00:55:53+08:00] INFO [dmsgpty:cli-server]: Cleanly stopped serving.
[2020-03-05T00:55:53+08:00] INFO [skywire]: CLI listener closed successfully
[2020-03-05T00:55:53+08:00] INFO [snet.dmsgC]: Cleanly stopped serving. dmsgpty="host" error="dmsg error 200 - local entity closed"
[2020-03-05T00:55:53+08:00] INFO [snet.dmsgC]: Serve() ended. dmsgpty="host" error=<nil>
[2020-03-05T00:55:53+08:00] WARN [rpc_server_02f9a04bf7d30c09c8b943735a1b3a1dc9f6ab9667bad1a2162a3df51cfce252be]: Error accepting conn on RPC Accept server side: listening on closed connection
[2020-03-05T00:55:53+08:00] INFO [skysocks]: [2020-03-05T00:55:53.328337109+08:00] DEBUG []: Failed to accept skysocks connection, but server is closed error="listening on closed connection"
[2020-03-05T00:55:53+08:00] INFO [proc_manager]: (skysocks) app stopped successfully
[2020-03-05T00:55:53+08:00] ERROR [proc_manager]: (skychat) failed to stop app error="os: process already finished"
[2020-03-05T00:55:53+08:00] INFO [router]: Closing all App connections and Loops
[2020-03-05T00:55:53+08:00] WARN [router]: setup client stopped serving error="dmsg error 200 - local entity closed"
[2020-03-05T00:55:53+08:00] DEBUG [proc_manager]: Remove app <skychat>: no such app
[2020-03-05T00:55:53+08:00] WARN [skywire]: App skychat stopped working: signal: interrupt
[2020-03-05T00:55:53+08:00] INFO [skynet]: Error accepting routing group: accept skynet: use of closed network connection
[2020-03-05T00:55:53+08:00] ERROR [skynet]: error serving error="accept skynet: use of closed network connection"
[2020-03-05T00:55:53+08:00] INFO [tp_manager]: transport manager is closing.
[2020-03-05T00:55:53+08:00] INFO [tp_manager]: transport manager closed.
[2020-03-05T00:55:53+08:00] WARN [tp_manager]: Failed to accept connection: io: read/write on closed pipe
[2020-03-05T00:55:53+08:00] WARN [skywire]: App skysocks stopped working: no such app
[2020-03-05T00:55:53+08:00] WARN [tp_manager]: Failed to accept connection: dmsg error 200 - local entity closed
[2020-03-05T00:55:53+08:00] INFO [skywire]: router stopped successfully
[2020-03-05T00:55:53+08:00] ERROR [skywire]: error serving RPC error="accept unix /tmp/visor_02ce0165ae0ce2e7bcacfcf15bd52ff4250e913b219f8fc27ca7ee656294b40ce2.sock: use of closed network connection"
[2020-03-05T00:55:53+08:00] ERROR [router]: Failed to read packet error="transport is no longer being served"
[2020-03-05T00:55:53+08:00] ERROR [app_rpc_server]: error closing conn error="close unix /tmp/visor_02ce0165ae0ce2e7bcacfcf15bd52ff4250e913b219f8fc27ca7ee656294b40ce2.sock->@: use of closed network connection"
[2020-03-05T00:55:53+08:00] INFO [skywire]: Socket file /tmp/visor_02ce0165ae0ce2e7bcacfcf15bd52ff4250e913b219f8fc27ca7ee656294b40ce2.sock removed successfully
```
Some of these should not be classed as `WARN` or `ERROR`.

### PR DESCRIPTION
**Pull request by evanlinjin**
Fixes #195 , Fixes #196

 Changes:	
* Made rpcutil.LogCall to log RPC call on the server side.
* Improved appserver.RPCGateway logging #195
* Changed appcommon.Key to use UUID.

### CODE REVIEW

File: pkg/app/appserver/rpc_gateway.go [hunk 1]
```diff
@@ -9,6 +9,8 @@ import (
 
 	"github.com/SkycoinProject/skycoin/src/util/logging"
 
+	"github.com/SkycoinProject/skywire-mainnet/pkg/util/rpcutil"
+
```
Comment (nkryuchkov) [inline pkg/app/appserver/rpc_gateway.go]: Could you please remove the empty line?

File: pkg/util/rpcutil/log_call.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import (
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// LogCall is used to log an RPC call from the rpc.Server
+func LogCall(log logrus.FieldLogger, method string, in interface{}) func(out interface{}, err *error) {
+
+	// Just in case log is not set.
```
Comment (nkryuchkov) [inline pkg/util/rpcutil/log_call.go:12]: Are you planning to fix a possible race condition or leave it as it is?

File: pkg/util/rpcutil/log_call.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	start := time.Now()
+	log = log.
+		WithField("_method", method).
+		WithField("_received", start.Format(time.Kitchen))
+	if in != nil {
+		log = log.WithField("input", in)
+	}
+
+	return func(out interface{}, err *error) {
+		log := log.WithField("_period", time.Since(start).String())
```
Comment (nkryuchkov) [inline pkg/util/rpcutil/log_call.go]: I would replace `period` with something like `elapsed`.

File: pkg/visor/rpc.go [hunk 4]
```diff
@@ -14,6 +14,8 @@ import (
 	"github.com/google/uuid"
 	"github.com/sirupsen/logrus"
 
+	"github.com/SkycoinProject/skywire-mainnet/pkg/util/rpcutil"
+
```
Comment (nkryuchkov) [inline pkg/visor/rpc.go]: Please remove the new line.

**Review by nkryuchkov — COMMENTED** [review]
Can we use new Go 1.13 errors features instead of `strings.Contains(err.Error(), ...)`? (I think it's `Unwrap()`).

File: pkg/util/rpcutil/log_call.go (hunk 2 above)
Comment (evanlinjin) [reply pkg/util/rpcutil/log_call.go:12]: Actually there wouldn't be any race conditions. I forgot to remove this comment.

**Comment by evanlinjin** [comment]
> strings.Contains

However it is only valid if we have implemented an `Unwrap()` member with all our `error` implementations. However, it's definitely something we need to look into.

**Pull request merged**